### PR TITLE
Fix harvesters refusing to allow queuing up docking orders when invalid

### PR DESCRIPTION
**Pull request by PunkPun**
It's a bug as by the time the queued dock order could be played as an activity, it might have become valid

Reported on discord

### CODE REVIEW

**Comment by penev92** [comment]
The original report: [message](https://discord.com/channels/153649279762694144/279290580201373696/1311082137423708263)

**Review by michaeldgg2 — APPROVED** [review]

File: OpenRA.Mods.Common/Activities/MoveToDock.cs [hunk 1]
```diff
@@ -29,6 +29,8 @@ public class MoveToDock : Activity
 		readonly bool forceEnter;
 		readonly bool ignoreOccupancy;
 
+		bool dockCancelled;
```
Comment (penev92) [inline OpenRA.Mods.Common/Activities/MoveToDock.cs]:
```suggestion
		bool isDockingCancelled;
```
please

**Review by penev92 — DISMISSED** [review]
LGTM other than that small naming thing.

**Review by penev92 — APPROVED** [review]
Almost how I wanted it

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/75f0303c135abd414c538708fc9665c5f074804e...c835f93d6671935d8e112e09b16374a04c20a87a)

Commits on `prep-2411`:
https://github.com/OpenRA/OpenRA/commit/52102da839fea895c4c78ce8ac4aa5fd6958588f
https://github.com/OpenRA/OpenRA/commit/2ccfa425b2cdd386ebc795491a729d48d867c8eb

**Pull request merged**